### DISCUSSION TITLE
Why there are 4 services created with standalone redis instance

### DISCUSSION THREAD

**Discussion by maheshglm** [OT-CONTAINER-KIT/redis-operator#857] [Q&A]
**Describe your question**
I am exploring operator based redis installation for my professional use, I have deployed redis-operator successfully and able to deploy `redis` standalone instance, and able to connect redis database without any issue. But I just want to understand why there are `4 services` created for single standalone instance. Also, please advise what is the recommended way to connect (i mean which service) from inside the cluster and from outside the cluster (I hope I can use external-service if I use load balancer).
And, I hope I can use Traefik ingress to expose the service instead of load balancer ? 

Btw, I am running this in GKE.

**What version of redis-operator are you using?**

redis-operator version: 0.15.1

**Additional context**

```
k get svc -n mgmt | grep redis
redis-infra-dev                                         ClusterIP      10.23.11.7     <none>        6379/TCP                     19m
redis-infra-dev-additional                              ClusterIP      10.23.9.248    <none>        6379/TCP                     19m
redis-infra-dev-external-service                        NodePort       10.23.11.103   <none>        6379:31758/TCP               19m
redis-infra-dev-headless                                ClusterIP      None           <none>        6379/TCP                     19m
```

Thanks in advance.

**Comment by drivebyer** [comment]
In my experience, when deploying standalone, we use redis-infra-dev to connect inside the cluster. For connections outside the cluster, we opt for redis-infra-dev-external-service.
